airflow/ui/src/components/DataTable/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect } from "react";
+
+type DagParams = {
+  configJson: string;
+  dagId: string;
+  logicalDate: string;
+  runId?: string;
+};
+
+type TriggerDAGFormProps = {
+  dagParams: DagParams;
+  onClose: () => void;
+  onTrigger: () => void;
+  setDagParams: React.Dispatch<React.SetStateAction<DagParams>>;
+};
+
+const TriggerDAGForm: React.FC<TriggerDAGFormProps> = ({
+  dagParams,
+  onTrigger,
+  setDagParams,
+}) => {
+  const [showDetails, setShowDetails] = useState(false);
+
+  useEffect(() => {
+    try {
+      const prettyJson = JSON.stringify(
+        JSON.parse(dagParams.configJson),
+        undefined,
+        2,
+      );
+
+      setDagParams((prev) => ({ ...prev, configJson: prettyJson }));
+    } catch {
+      // Invalid JSON handling
+    }
+  }, [dagParams.configJson, setDagParams]);
+
+  const handleChange = (
+    ele: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = ele.target;
+
+    setDagParams((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleReset = () => {
+    setDagParams({
+      configJson: "{}",
+      dagId: dagParams.dagId,
+      logicalDate: "",
+      runId: "",
+    });
+  };
+
+  const handleJsonChange = (value: string) => {
+    setDagParams((prev) => ({ ...prev, configJson: value }));
+  };
+
+  const isValidJson = () => {
+    try {
+      JSON.parse(dagParams.configJson);
+
+      return true;
+    } catch {
+      return false;
+    }
+  };
+
+  return (
+    <>
+      <ModalCloseButton />
+
+      <VStack align="stretch" p={5} spacing={2}>
+        <Button
+          onClick={() => setShowDetails(!showDetails)}
+          variant="outline"
+          width="full"
+        >
+          {showDetails ? "Hide Advance Options" : "Show Advance Options"}
+        </Button>
+
+        <Collapse in={showDetails}>
+          <VStack align="stretch" spacing={3}>
+            <FormControl>
+              <FormLabel size="sm">Logical date</FormLabel>
+              <Input
+                name="logicalDate"
+                onChange={handleChange}
+                placeholder="yyyy-mm-ddThh:mm"
+                size="sm"
+                type="datetime-local"
+                value={dagParams.logicalDate}
+              />
+            </FormControl>
+
+            <FormControl>
+              <FormLabel size="sm">Run ID (Optional)</FormLabel>
+              <Input
+                name="runId"
+                onChange={handleChange}
+                placeholder="Autogenerated if left empty"
+                size="sm"
+                value={dagParams.runId}
+              />
+            </FormControl>
+
+            <FormControl>
+              <FormLabel size="sm">Configuration JSON</FormLabel>
+              <CodeMirror
+                extensions={[json(), autocompletion(), lineNumbers()]}
+                height="200px"
+                onChange={handleJsonChange}
```

Comment:
While I was typing JSON text the cursor jumped around and then then JSON was malformed. I assume the native auto-formatting of CodeMirror is better than this.

Reply:
Yes, I edited most of them to codemirror, some functionalities is yet to do. Will add them under extension.